### DISCUSSION TITLE
How can I create or assign a role admin to a user

### DISCUSSION THREAD

**Discussion by AbdelilahDerfoufi** [symfony/symfony#44922] [Q&A]
I am developing an app with symfony 5.4 and I can't figure out how can I assign or create user with role admin, and how can I do it through easyadmin bundle?

**Comment by ghost** [comment]
I think you should ask this question in EAB discussions if they have their own way to manage roles.
In common Symfony application you just create a CRUD functionality to manage user roles (or embed this in user crud).

**Reply by AbdelilahDerfoufi** [reply]
Then should I change my approach? how can I handle roles with a basic CRUD?

**Reply by ghost** [reply]
It depends on your entities design. The simplest it to create a form with a multiselect field on the frontend and a backend process to handle submission. How you will handle this depends also on how store the roles i.e. as strings in a field in user object or a separate entities with a user owning a collection (probably many to many relation) or some other way.

**Reply by AbdelilahDerfoufi** [reply]
The roles field in my user class is the default roles in json that comes when typing make:user, what about this case here?

**Reply by AbdelilahDerfoufi** [reply]
I used ChoiceType to woth roles as options, but the error says that  ChoiceType class doesn't exist and could not load

**Reply by ghost** [reply]
> I used ChoiceType to woth roles as options, but the error says that ChoiceType class doesn't exist and could not load

This sentence is mutually exclusive. First you say you had used ChoiceType and then you say you had got an error.

**Reply by AbdelilahDerfoufi** [reply]
It was a spelling error in the use statement, sorry hhh.
But there is the problem of hashing the password I think, because I create a user but in the login form I get invalid token

**Reply by ghost** [reply]
What token? Please be more verbose. CSRF token? Did you render this fileld (which is a hidden field)? Do you have csrf protection enabled?
The following documentation might be helpful:
https://symfony.com/doc/current/security/csrf.html

**Reply by AbdelilahDerfoufi** [reply]
![image](https://user-images.githubusercontent.com/49759161/149137688-7e464420-da1f-4689-8494-224b5351b4f8.png)

I meant Invalid credentials., sorry

**Reply by AbdelilahDerfoufi** [reply]
I followed a tutorial but I have an issue with this interface so I couldn't complete it.

![image](https://user-images.githubusercontent.com/49759161/149138010-74306ab6-522a-45e8-894f-906339c53da7.png)

And here is my user:

![image](https://user-images.githubusercontent.com/49759161/149138260-a72860e2-02af-4b44-90c5-abfe83247b04.png)

**Reply by ghost** [reply]
> but I have an issue with this interface

I'm sorry but I can't help you any more. Please learn how to describe problems first. Saying "I have an issue" means absolutely nothing. If you want help from other developers you have to be much more explicit.

**Reply by AbdelilahDerfoufi** [reply]
I'm sorry, but I didn't know how to describe, I get a message that says UserPasswordEncoderInterface is deprecated and I don't know what to do?

**Reply by norkunas** [reply]
https://symfony.com/blog/new-in-symfony-5-3-passwordhasher-component use PasswordHasher :)

**Reply by AbdelilahDerfoufi** [reply]
Would this help me help encode password when creating a new user or editing it

(Edit) It worked
![image](https://user-images.githubusercontent.com/49759161/149161212-a9a609d9-d2c0-410c-8d35-e808024c88b8.png)
